### PR TITLE
[CONFIGURATION-764] Default date lookup can not work for some specific

### PR DESCRIPTION
**Pull request by NingZhang-e**
Fix for https://issues.apache.org/jira/browse/CONFIGURATION-764

### CODE REVIEW

**Comment by garydgregory** [comment]
@NingZhang-e May you please rebase on master?

**Comment by NingZhang-e** [comment]
@garydgregory Rebased.

File: src/changes/changes.xml
Comment (garydgregory) [inline src/changes/changes.xml:164]: No extra whitespace.

File: src/test/resources/testDefaultLookups.xml
Comment (garydgregory) [inline src/test/resources/testDefaultLookups.xml:23]: Extra whitespace

File: src/test/java/org/apache/commons/configuration2/interpol/TestDefaultLookups.java
Comment (garydgregory) [inline src/test/java/org/apache/commons/configuration2/interpol/TestDefaultLookups.java:32]: Extra whitespace

File: src/main/java/org/apache/commons/configuration2/interpol/StringLookupCatchExceptionAdapter.java
Comment (garydgregory) [inline src/main/java/org/apache/commons/configuration2/interpol/StringLookupCatchExceptionAdapter.java:24]: Wrong since version.

**Review by garydgregory — CHANGES_REQUESTED** [review]
Thank you for your PR, please see comments. Needs further review from the community.

File: src/changes/changes.xml
Comment (NingZhang-e) [reply src/changes/changes.xml]: Done

File: src/test/resources/testDefaultLookups.xml
Comment (NingZhang-e) [reply src/test/resources/testDefaultLookups.xml]: Done

File: src/test/java/org/apache/commons/configuration2/interpol/TestDefaultLookups.java
Comment (NingZhang-e) [reply src/test/java/org/apache/commons/configuration2/interpol/TestDefaultLookups.java:32]: I did not get it. Just same as other file.

File: src/main/java/org/apache/commons/configuration2/interpol/StringLookupCatchExceptionAdapter.java
Comment (NingZhang-e) [reply src/main/java/org/apache/commons/configuration2/interpol/StringLookupCatchExceptionAdapter.java]: Done

**Comment by NingZhang-e** [comment]
> Thank you for your PR, please see comments. Needs further review from the community.
Thanks for review. When the community could review it? 
This PR has been here almost 2 years, I nearly forgot it and never expect someone could look at it. :-)

File: src/main/java/org/apache/commons/configuration2/interpol/StringLookupCatchExceptionAdapter.java
Comment (kinow) [inline src/main/java/org/apache/commons/configuration2/interpol/StringLookupCatchExceptionAdapter.java:42]: Formatting issues above :point_up:

File: src/changes/changes.xml
Comment (kinow) [inline src/changes/changes.xml:51]: You can leave `changes.xml` to be updated later by a committer, no need to include in this PR :+1: (it's also conflicting with `master` now, so removing it from this PR will fix that).

File: src/main/java/org/apache/commons/configuration2/interpol/StringLookupCatchExceptionAdapter.java
Comment (kinow) [inline src/main/java/org/apache/commons/configuration2/interpol/StringLookupCatchExceptionAdapter.java:26]: Maybe `StringLookupAdapter` could have a parameter like `suppress=false` that's disabled by default? This class is not public, so either way is fine, we can improve on it later IMO.

**Review by kinow — CHANGES_REQUESTED** [review]
Couple comments. Thanks @NingZhang-e !

File: src/main/java/org/apache/commons/configuration2/interpol/StringLookupCatchExceptionAdapter.java
Comment (darkma773r) [reply src/main/java/org/apache/commons/configuration2/interpol/StringLookupCatchExceptionAdapter.java:26]: One issue I see here is that if we suppress exceptions in the adapter, then legitimate configuration errors will not be propagated. For example, say I have a property like `current date= ${date:not-a-format}`. With the changes here, this would interpolate without error to the string `current date= ${date:not-a-format}` instead of throwing an exception about the invalid format string, as is the current behavior.

**Comment by darkma773r** [comment]
I've submitted #182, with an alternative fix for this that updates the single variable detection logic instead of suppressing errors.

File: src/test/java/org/apache/commons/configuration2/interpol/TestDefaultLookups.java
Comment (robtimus) [reply src/test/java/org/apache/commons/configuration2/interpol/TestDefaultLookups.java:32]: I think @garydgregory was referring to line 31.

**Comment by darkma773r** [comment]
Closing this PR. The functionality was implemented with #182 and released in 2.8.0.

**Pull request closed without merging**
